apps/dot_ext/oauth2_backends.py

```diff
@@ -21,7 +22,11 @@ def create_token_response(self, request):
 
         headers, body, status = self.server.create_token_response(uri, http_method, body,
                                                                   headers, extra_credentials)
+        if (status >= 300):
+            raise APIException(detail=body, code=status)
```

Comment:
I have some concerns about this. First, it's not entirely clear what's in body here, and I'm worried about blindly returning a blob of data.

In addition, handling 3XX statuses in the same way as 4XX and 5XX statuses seems very problematic. I'm not sure there's any way this can happen in the current implementation, but I'm pretty nervous about lumping them all together.

Reply:
I'm not sure I understand what you mean about lumping 3xx etc together, it would be easy to add but what kind of handling behavior would you have that differed between the three?

Body in this case would contain the error from `create_token_response`, but yeah I agree ( missing typed languages right now -_- ) I'd say we could make a generic body but the information is valuable to pass back to the user so they can attempt to rectify the problem. Also since it's coming from a function owned by a library we're trusting with our authentication it's probably ok (meaning we're already trusting it with more important things).